src/sockperf.cpp

```diff
@@ -2304,6 +2323,21 @@ void cleanup() {
                 if (g_fds_array[ifd]->is_multicast) {
                     FREE(g_fds_array[ifd]->memberships_addr);
                 }
+                if (s_user_params.addr.ss_family == AF_UNIX) {
+                    unlink(s_user_params.client_bind_info.addr_un.sun_path);
+                    if (s_user_params.mode == MODE_CLIENT && s_user_params.sock_type == SOCK_DGRAM) { // unlink binded client
+                        std::string sun_path = s_user_params.addr.addr_un.sun_path;
+                        char pid_buffer [MAX_FDS_NUM];
+                        char ifd_buffer [MAX_FDS_NUM];
+                        snprintf(pid_buffer, sizeof(pid_buffer), "%d", getpid());
+                        snprintf(ifd_buffer, sizeof(ifd_buffer), "%d", ifd);
+                        sun_path = sun_path + "_" + pid_buffer + "_" + ifd_buffer;
```

Comment:
You're using similar code in `Client::initBeforeLoop()`. Please extract to a common function.